File: date_utc/java_date_utc/DateUtc.java
import java.lang.*;
import java.util.*;
import java.text.*;
import java.time.format.*;
import java.time.*;

public class DateUtc {
  public static void main( String args[] ) {
    if( args.length != 0 ) {
      System.err.println( "Usage: java -jar DateUtc.jar <noargs>");
      System.exit(1);
    }
    date_utc();
  }

  public static void date_utc() {
    int year;
    int month;
    int day;
    int hour;
    int min;
    int sec;

    LocalDateTime local = LocalDateTime.now();
    year = local.getYear();
    month = local.getMonthValue();
    day = local.getDayOfMonth();
    hour = local.getHour();
    min = local.getMinute();
    sec = local.getSecond();

    System.out.println(DateTimeFormatter.ofPattern("yyyy-MM-dd hh:mm:ss").format(local));
    System.out.println("lccal=<" + year + "-" + month + "-" + day + " " + hour + ":" + min + ":" + sec + ">");

    ZonedDateTime utc = ZonedDateTime.now(ZoneOffset.UTC);
    year = utc.getYear();
    month = utc.getMonthValue();
    day = utc.getDayOfMonth();
    hour = utc.getHour();
    min = utc.getMinute();
    sec = utc.getSecond();
    System.out.println(DateTimeFormatter.ofPattern("yyyy-MM-dd hh:mm:ss").format(utc));
    System.out.println("utc=<" + year + "-" + month + "-" + day + " " + hour + ":" + min + ":" + sec + ">");
  }
}

File: date_utc/kotlin_date_utc/DateUtc.kt
package DateUtc

import java.text.SimpleDateFormat
import java.util.*

class DateUtc {
  fun date_utc_function(n : Int) : Int {
    return n
  }


  fun String.toDate(dateFormat: String = "yyyy-MM-dd HH:mm:ss", timeZone: TimeZone = TimeZone.getTimeZone("UTC")): Date {
val parser = SimpleDateFormat(dateFormat, Locale.getDefault())
parser.timeZone = timeZone
return parser.parse(this)
}

  fun Date.formatTo(dateFormat: String, timeZone: TimeZone = TimeZone.getDefault()): String {
    val formatter = SimpleDateFormat(dateFormat, Locale.getDefault())
    formatter.timeZone = timeZone
    return formatter.format(this)
  }
}



fun main(args : Array<String>) {
  if( args.size != 0 ) {
    println("Usage: java -jar DateUtc.jar <noargs>")
    System.exit(1)
  }
  val date_utc = DateUtc()
  for (idx_i in 1..9) {
    println("date_utc_function(" + idx_i + ")=<" + date_utc.date_utc_function(idx_i) + ">")
  }
}

